src/clients/HubPoolClient.ts

```diff
@@ -107,7 +107,7 @@ export class HubPoolClient extends BaseAbstractClient {
     cachingMechanism?: CachingMechanismInterface
   ) {
     super(cachingMechanism);
-    this.latestBlockNumber = deploymentBlock === 0 ? deploymentBlock : deploymentBlock - 1;
+    this.latestBlockSearched = deploymentBlock === 0 ? deploymentBlock : deploymentBlock - 1;
```

Comment:
Opportunistic tweak:

```suggestion
    this.latestBlockSearched = Math.min(deploymentBlock - 1, 0);
```